src/main/java/com/minecolonies/client/gui/WindowHutFarmer.java

```diff
@@ -201,37 +190,122 @@ public String getBuildingName()
         return "com.minecolonies.gui.workerHuts.farmer";
     }
 
+    /**
+     * Retrieve levels from the building to display in GUI
+     */
+    private void pullLevelsFromHut()
+    {
+        fields = building.getFields();
+    }
+
     @Override
     public void onOpened()
     {
         super.onOpened();
 
-        updateButtonLabels();
+        if(building.assignFieldManually())
+        {
+            findPaneOfTypeByID(TAG_BUTTON_ASSIGNMENT_MODE, Button.class).setLabel(LanguageHandler.format("com.minecolonies.gui.hiring.on"));
+        }
+        else
+        {
+            findPaneOfTypeByID(TAG_BUTTON_ASSIGNMENT_MODE, Button.class).setLabel(LanguageHandler.format("com.minecolonies.gui.hiring.off"));
+        }
+
         findPaneOfTypeByID(BUTTON_PREV_PAGE, Button.class).setEnabled(false);
         buttonPrevPage = findPaneOfTypeByID(BUTTON_PREV_PAGE, Button.class);
         buttonNextPage = findPaneOfTypeByID(BUTTON_NEXT_PAGE, Button.class);
+
+        fieldList = findPaneOfTypeByID(LIST_FIELDS, ScrollingList.class);
+        fieldList.setDataProvider(new ScrollingList.DataProvider()
+        {
+            @Override
+            public int getElementCount()
+            {
+                return fields.size();
+            }
+
+            @Override
+            public void updateElement(int index, Pane rowPane)
+            {
+                FieldView field = fields.get(index);
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Local variable 'field' could be declared final [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=pmd%3ALocalVariableCouldBeFinal)